process/block/preprocess/selectionSession.go

```diff
@@ -38,67 +39,80 @@ func NewSelectionSession(args ArgsSelectionSession) (*selectionSession, error) {
-		return nil, err
+		// Unexpected failure.
+		return 0, nil, false, err
 	}
-
-	userAccount, ok := account.(state.UserAccountHandler)
-	if !ok {
-		return nil, process.ErrWrongTypeAssertion
+	if check.IfNil(account) {
+		// New (unknown) account.
+		return 0, big.NewInt(0), false, nil
 	}
 
-	return userAccount, nil
+	return account.GetNonce(), account.GetBalance(), true, nil
 }
 
-func (session *selectionSession) getExistingAccount(address []byte) (vmcommon.AccountHandler, error) {
+func (session *selectionSession) getCachedUserAccount(address []byte) (state.UserAccountHandler, error) {
 	account, ok := session.ephemeralAccountsCache[string(address)]
 	if ok {
+		// Existing or new (unknown) account, previously-cached.
 		return account, nil
 	}
 
-	account, err := session.accountsAdapter.GetExistingAccount(address)
-	if err != nil {
+	account, err := session.getExistingAccountTypedAsUserAccount(address)
+	if err != nil && err != state.ErrAccNotFound {
+		// Unexpected failure (error different from "ErrAccNotFound").
+		// Account won't be cached.
 		return nil, err
 	}
 
+	// Existing account or new (unknown), we'll cache it (actual object or nil).
 	session.ephemeralAccountsCache[string(address)] = account
+
+	// Generally speaking, this isn't a good pattern: returning both nil (for unknown accounts), and a nil error.
+	// However, this is a non-exported method, which should only be called with care, within this struct only.
 	return account, nil
 }
 
-// IsIncorrectlyGuarded checks if a transaction is incorrectly guarded (not executable).
-// Will be called by mempool during transaction selection.
-func (session *selectionSession) IsIncorrectlyGuarded(tx data.TransactionHandler) bool {
-	address := tx.GetSndAddr()
-	account, err := session.getExistingAccount(address)
+func (session *selectionSession) getExistingAccountTypedAsUserAccount(address []byte) (state.UserAccountHandler, error) {
+	account, err := session.accountsAdapter.GetExistingAccount(address)
 	if err != nil {
-		return false
+		return nil, err
 	}
 
 	userAccount, ok := account.(state.UserAccountHandler)
 	if !ok {
-		// On this branch, we are (approximately) mirroring the behavior of "transactionsProcessor.VerifyGuardian()".
-		return true
+		return nil, process.ErrWrongTypeAssertion
+	}
+
+	return userAccount, nil
+}
+
+// IsIncorrectlyGuarded checks if a transaction is incorrectly guarded (not executable).
+// Will be called by mempool during transaction selection.
+func (session *selectionSession) IsIncorrectlyGuarded(tx data.TransactionHandler) bool {
+	address := tx.GetSndAddr()
+	account, err := session.getCachedUserAccount(address)
+	if err != nil || check.IfNil(account) {
+		// Unexpected failure.
+		// In case of any error, we assume the transaction is "correctly guarded".
+		// This includes the error "ErrAccNotFound", when the account is obviously not guarded: this special case is denied by interceptors beforehand.
+		return false
```

Comment:
Maybe it's best to return _false_ here.

Generally speaking, this is ~dead code.